svn/common.py

```diff
@@ -28,7 +29,7 @@ def __init__(self, url_or_path, type_, *args, **kwargs):
         self.__trust_cert = kwargs.pop('trust_cert', None)
 
         if type_ not in (svn.constants.LT_URL, svn.constants.LT_PATH):
-            raise ValueError("Type is invalid: %s" % (type_))
+            raise SvnException("Type is invalid: %s" % type_)
```

Comment:
It's not Python's convention to use new exceptions where existing
exceptions will suffice. We should just use the new exception where we
can't sufficiently use one of the well-known ones. For example, we
shouldn't use ValueError from the lower levels because it could take on
multiple interpretations. However, someone passing a value that does not
exist within the discrete set of potential choices would understand exactly
what a ValueError was trying to tell them.

Plus, given its name, an SvnException can be used just to describe SVN
command failures.

Dustin

On Sep 14, 2016 06:10, "Tushar Makkar" <notifications@github.com> wrote:

> In svn/common.py
> <https://github.com/dsoprea/PySvn/pull/56#discussion_r78720925>:
>
> > @@ -28,7 +29,7 @@ def __init__(self, url_or_path, type_, *args, **kwargs):
> >          self.__trust_cert = kwargs.pop('trust_cert', None)
> >
> >          if type_ not in (svn.constants.LT_URL, svn.constants.LT_PATH):
> > -            raise ValueError("Type is invalid: %s" % (type_))
> > +            raise SvnException("Type is invalid: %s" % type_)
>
> I was thinking of having SvnException as the parent exception of all the
> exception's raised by PySvn. @dsoprea <https://github.com/dsoprea> What
> do you think about this? According to me, we should have custom exceptions
> only and the solution here is to update the docstring for providing more
> apt information.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dsoprea/PySvn/pull/56/files/95e67ad6b9b447bbd3105de50e551a04003b7c6e#r78720925>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AArratdVDIyYhsu_z0ETPK8rjn3Xditqks5qp8gigaJpZM4J7z3r>
> .
>